test/test_binary_ufuncs.py

```diff
@@ -1425,31 +1425,34 @@ def test_fmod(self, device, dtype):
         def _reference_implementation(x, mod):
             np_x = x.cpu().numpy()
             np_mod = 0
-            # No type promotion
-            # Issue #47779: https://github.com/pytorch/pytorch/issues/47779
             if torch.is_tensor(mod):
                 np_mod = mod.cpu().numpy()
             else:
                 np_mod = mod
-                # Non XLA platform needs to cast to int
-                if dtype in torch.testing.get_all_int_dtypes() and self.device_type in ['cpu', 'cuda']:
-                    np_mod = int(np_mod)
             exp = np.fmod(np_x, np_mod)
             exp = torch.from_numpy(exp)
 
             res = torch.fmod(x, mod)
-            res = res.to(exp.dtype)
+            common_dtype = torch.promote_types(res.dtype, exp.dtype)
```

Comment:
This seems a little wonky since part of the test's goal is to validate the correct dtype is returned. Before altering the dtype of res, what about comparing its dtype to the result of promote_types?

Reply:
The reason I promote the result and expect to same result is the divergence of type promotion between torch and numpy:
1. All integral tensors  x  float32 scalar
    - numpy: float64
    - torch: float32
2. int32/int64 tensors  x  float32 tensor
    - numpy: float64
    - torch: float32
So, I decide to add these cases to use `exact_dtype=False`, and get rid of type promotion over result.